### ISSUE TITLE
Allow to create spider with custom templates

### ISSUE THREAD

**Issue by drjackild** [scrapy/scrapy#2046]
I don't know, am I wrong, but is it possible now to create custom templates and create spiders with `scrapy genspider -t <custom_template> <name> <domain>`? As I see in source code, user can set custom template folder in TEMPLATE_FOLDER variable, but is there any documentation about template language, which using in template? 

Actually, I've planned to work on PR with possibilities to create custom templates, but probably it's already implemented. May be just add documentation about this feature?

UPD. Ah, I see, that `_genspider` using `render_templatefile`, which just using `string.Template().substitute()` to render template and save .py file

**Comment by kmike** [comment]
Hey @DrJackilD,

You can use http://cookiecutter.readthedocs.io/en/latest/ to generate projects based on custom templates; it is more powerful and could be easier to use than Scrapy's template generating facilities.

**Comment by DrJackilD** [comment]
@kmike yeah, that's a great tool, but I'm talking about spider's level. or cookiecutter can be used in this direction too?

**Comment by kmike** [comment]
Ah, sorry, I missed that; if you want to generate a single file than Scrapy's engine should be enough, cookiecutter looks like an overkill. So yes, creating a folder, pointing TEMPLATES_DIR option to it and storing templates there makes sense. This is kind of documented in [TEMPLATES_DIR](http://doc.scrapy.org/en/latest/topics/settings.html?highlight=templates_dir#templates-dir) docs, but you're right this is not explained well in neither in [genspider](http://doc.scrapy.org/en/latest/topics/commands.html#std:command-genspider) docs nor in genspider help string.

**Comment by DrJackilD** [comment]
@kmike Ok, I'll make PR with some explanation about .tmpl files and necessary arguments, which need to be in templates. Thanks for the discussion :)

**Comment by 3rawkz** [comment]
overkill?! hmmmm... negative... a must for any pyHead... but yeah, for the sole purpose of scrapy templates? lol waste  ... like being at a buffet and and watch your figure, why there in first place? Hope you added it to your libs anyhow!!!

**Comment by Congee** [comment]
@DrJackilD ping
